Amazon.QLDB.Driver/transaction/IAsyncTransaction.cs

```diff
@@ -0,0 +1,28 @@
+﻿/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
+ * the License. A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+namespace Amazon.QLDB.Driver
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    internal interface IAsyncTransaction : IAsyncDisposable, IAsyncExecutable
```

Comment:
I don't think we need this class.

Reply:
I kept it for the sake of symmetry with non-async.
ITransaction defines void methods while IAsyncTransaction defined methods taking in a cancellation token and returning Task.

Reply:
I still think we should remove it. The only reason we have an ITransaction is because of legacy code back when Transaction was exposed to the consumer.

Reply:
Removed.